libraries/src/Form/Field/FilelistField.php

```diff
@@ -187,17 +187,15 @@ public function setup(\SimpleXMLElement $element, $value, $group = null)
 	 */
 	protected function getOptions()
 	{
-		$options = array();
+		$options = [];
 
-		$path = $this->directory;
+		$path = Path::clean(strpos($this->directory, '/') === 0 ? $this->directory : (JPATH_ROOT . '/' . $this->directory));
```

Comment:
no offense but I'm always impressed how unreadable code can be ;-)

```suggestion
		$path = $this->directory;
		
		if (strpos($path, '/') === 0)
		{
			$path = JPATH_ROOT . '/' . $path;
		}
		
		$path = Path::clean($path);
```